Framework/PythonInterface/mantid/plots/__init__.py

```diff
@@ -611,12 +619,33 @@ def plot(self, *args, **kwargs):
 
             def _data_update(artists, workspace, new_kwargs=None):
                 # It's only possible to plot 1 line at a time from a workspace
+                try:
+                    if new_kwargs:
+                        x, y, _, __ = plotfunctions._plot_impl(self, workspace, args, new_kwargs)
+                    else:
+                        x, y, _, __ = plotfunctions._plot_impl(self, workspace, args, kwargs)
+                    artists[0].set_data(x, y)
+                except RuntimeError as ex:
+                    # if curve couldn't be plotted then remove it - can happen if the workspace doesn't contain the
+                    # spectrum any more following execution of an algorithm
+                    logger.information('Curve not plotted: {0}'.format(ex.args[0]))
+
+                    # remove the curve using similar to logic that in _WorkspaceArtists._remove
+                    artists[0].remove()
+
+                    # blank out list that will be returned
+                    artists = []
+
+                    # also remove the curve from the legend
+                    if (not self.is_empty(self)) and self.legend_ is not None:
+                        self.legend().draggable()
+
                 if new_kwargs:
-                    x, y, _, __ = plotfunctions._plot_impl(self, workspace, args, new_kwargs)
+                    _autoscale = new_kwargs.pop("autoscale_on_update", self.get_autoscale_on())
                 else:
-                    x, y, _, __ = plotfunctions._plot_impl(self, workspace, args, kwargs)
-                artists[0].set_data(x, y)
-                if new_kwargs and new_kwargs.pop('autoscale_on_update', self.get_autoscale_on()):
+                    _autoscale = self.get_autoscale_on()
+
+                if _autoscale:
```

Comment:
```suggestion
                if autoscale_enabled:
```

Or something to indicate this controls if auto scaling happens.

Reply:
OK I've renamed this variable to _autoscale_on which occurred in 2 places in this unit. I have also renamed a similar autoscale variable in the same unit